src/hera/workflows/_mixins.py

```diff
@@ -574,6 +574,9 @@ class ArgumentsMixin(BaseMixin):
 
     def _build_arguments(self) -> Optional[ModelArguments]:
         """Processes the `arguments` field and builds the optional generated `Arguments` to set as arguments."""
+        # Reapply the validator in case arguments was assigned to as a dictionary
+        self.arguments = normalize_to_list_or(ModelArguments)(self.arguments)
```

Comment:
Should we maybe assign this to a local variable instead of modifying `self.arguments`? Changing the field as a side-effect is potentially surprising, and didn't happen here before (and side-effects can potentially lead to bugs later on).